packages/ses/src/strict-scope-terminator.js

```diff
@@ -0,0 +1,92 @@
+import {
+  Proxy,
+  String,
+  TypeError,
+  ReferenceError,
+  create,
+  freeze,
+  getOwnPropertyDescriptors,
+  globalThis,
+  immutableObject,
+} from './commons.js';
+import { assert } from './error/assert.js';
+
+const { details: d, quote: q } = assert;
+
+/**
+ * alwaysThrowHandler
+ * This is an object that throws if any property is called. It's used as
+ * a proxy handler which throws on any trap called.
+ * It's made from a proxy with a get trap that throws. It's safe to
+ * create one and share it between all Proxy handlers.
+ */
+export const alwaysThrowHandler = new Proxy(
+  immutableObject,
+  freeze({
+    get(_shadow, prop) {
+      // eslint-disable-next-line @endo/no-polymorphic-call
+      assert.fail(
+        d`Please report unexpected scope handler trap: ${q(String(prop))}`,
+      );
+    },
+  }),
+);
+
+/*
+ * scopeProxyHandlerProperties
+ * scopeTerminatorHandler manages a strictScopeTerminator Proxy which serves as
+ * the final scope boundary that will always return "undefined" in order
+ * to prevent access to "start compartment globals".
+ */
+const scopeProxyHandlerProperties = {
+  get(_shadow, _prop) {
+    return undefined;
+  },
+
+  set(_shadow, prop, _value) {
+    // we should only hit this if the has() hook returned true
+    // matches the v8 ReferenceError message
+    // "Uncaught ReferenceError: xyz is not defined"
+    throw new ReferenceError(`${String(prop)} is not defined`);
+  },
+
+  has(_shadow, prop) {
+    // we must at least return true for all properties on the realm globalThis
```

Comment:
Is that still true though? 
All of them are already handled in the previous backflip. 
This is really only needed for sloppy mode, which could be overridden in there anyway, right? 
Strict scope terminator shouldn't need a reference to any globals I think.

Reply:
still true. in order to block access to the **realm globalThis** properties we need to return true in this has hook

Reply:
Herein lies the leak of information about the properties of the underlying realm’s globalThis, which we found to be less bad than breaking `typeof`. I would feel most confident about our security claims if this simply returned `true`. I like that this new formulation of the scope proxy makes this compromise more obvious!

Reply:
yes! i think we can `return true` here to no change in behavior

Reply:
this breaks a lot of tests, including tests outside of ses
> 17 tests failed
for example where it previously threw an error
```js
t.throws(() => c.evaluate('XYZ'), { instanceOf: ReferenceError });
```
it now just returns `undefined`
```js
t.is(c.evaluate('XYZ'), undefined);
```

i think we should consider this change outside of this PR

Reply:
Agreed. We’ll revisit this behavior change. #1305